src/utils/vvim.cpp

```diff
@@ -1029,6 +1029,15 @@ bool VVim::handleKeyPressEvent(int key, int modifiers, int *p_autoIndentPos)
             }
 
             setMode(VimMode::Insert);
+        } else if (modifiers == Qt::ShiftModifier) {
```

Comment:
As far as I know, `S` is the same as `cc`, right? I have implemented the `cc`. I think you could just copy that logic here. By the way, we need to care about the `[count]` repeat number and the auto-indent, which has been handled in the `cc`.